src/components/task-plan/reading-stats.cjsx

```diff
@@ -0,0 +1,117 @@
+  componentWillUnmount: -> TaskPlanStore.removeChangeListener(@update)
+
+  getPlanId: () ->
+    @getParams().id or @state.id
+
+  update: -> @setState {}
+
+  percent: (num,total) ->
+    Math.round((num/total) * 100)
+
+  percentDelta: (a,b) ->
+    if a > b
+      change = a - b
+      op = '+'
+    else if a == b
+      change = 0
+      op = ''
+    else
+      change = b - a
+      op = '-'
+    op + ' ' + Math.round((change / b) * 100)
+
+    
+  
+  renderCourseBar: (data) ->
+    max = data.complete_count + data.partially_complete_count + (data.total_count - (data.complete_count + data.partially_complete_count))
+    <div className="reading-progress-container">
+      <div className="reading-progress-heading">
+        complete / in progress / not started
+      </div>
+      <BS.ProgressBar className="reading-progress-group">
+        <BS.ProgressBar className="reading-progress-bar" bsStyle="success" label="%(now)s" max={max} now={data.complete_count} key={1} />
+        <BS.ProgressBar className="reading-progress-bar" bsStyle="warning" label="%(now)s" max={max} now={data.partially_complete_count} key={2} />
+        <BS.ProgressBar className="reading-progress-bar" bsStyle="danger" label="%(now)s" max={max} now={data.total_count - (data.complete_count + data.partially_complete_count)} key={3} />
+      </BS.ProgressBar>
+    </div> 
+
+  renderChapterBars: (data, i) ->
+    if data.correct_count > 0
+      correct = <BS.ProgressBar className="reading-progress-bar" bsStyle="success" label="%(percent)s%" now={@percent(data.correct_count,data.correct_count+data.incorrect_count)} key={1} />
+    else
+      correct = ' '
+    if data.incorrect_count > 0
+      incorrect = <BS.ProgressBar className="reading-progress-bar" bsStyle="warning" label="%(percent)s%" now={@percent(data.incorrect_count,data.correct_count+data.incorrect_count)} key={2} />
+    else
+      incorrect = ' '
+    <div>
+      <div className="reading-progress-heading">
+        {data.page.number} - {data.page.title}
+      </div>
+      <div className="reading-progress-container">
+        <BS.ProgressBar className="reading-progress-group">
+          {correct}
+          {incorrect}
+        </BS.ProgressBar>
+        
+      </div>
+      
+    </div>
+
+  renderPracticeBars: (data, i) ->
+    <div>
+      <div className="reading-progress-heading">
+        {data.page.number} - {data.page.title}
+      </div>
+      <div className="reading-progress-container">
+        <BS.ProgressBar className="reading-progress-group">
+          <BS.ProgressBar className="reading-progress-bar" bsStyle="success" label="%(now)s%" now={@percent(data.correct_count,data.correct_count+data.incorrect_count)} key={1} />
+        </BS.ProgressBar>
+        <div className="reading-progress-delta">{@percentDelta(data.correct_count,data.previous_attempt.correct_count)}% change</div>
+      </div>
+    </div>
+
+  render: ->
+    id = @getPlanId()
+
+    if TaskPlanStore.isLoaded(id)
```

Comment:
Did you try using the Loadable mixin to handle this case (and the `componentWill(Un)Mount`)?